src/datasets/download/streaming_download_manager.py

```diff
@@ -406,8 +406,12 @@ def _prepare_http_url_kwargs(url: str, use_auth_token: Optional[Union[str, bool]
     # Fix Google Drive URL to avoid Virus scan warning
     if "drive.google.com" in url and "confirm=" not in url:
         url += "&confirm=t"
-    if url.startswith("https://raw.githubusercontent.com/"):
+    if (
+        url.startswith("https://raw.githubusercontent.com/")
         # Workaround for served data with gzip content-encoding: https://github.com/fsspec/filesystem_spec/issues/389
+        or url.startswith("https://gitlab.lip6.fr/")
```

Comment:
Why is this change needed ? On my side reading the file from this website in streaming mode works (though it takes some time to download)

Reply:
It was not working on my side. I can recheck again...

Reply:
@lhoestq if I remove this line of code, I get this error in streaming mode:
```
BlockSizeError: Got more bytes so far (>5255168) than requested (5251072)
```
I don't think their server supports HTTP Range requests and these must be disallowed.

Reply:
From the master branch, if I load the new MLSum dataset script (on https://github.com/huggingface/datasets/pull/4574/commits/48f1775c312955519c0a469096ae74a6c5841bde) it works fine
```python
ds = load_dataset("mlsum", "fr", revision="48f1775c312955519c0a469096ae74a6c5841bde", streaming=True)
next(iter(ds["train"]))
```
with fsspec==2022.3.0

What code do you run to get an error ?

Reply:
@lhoestq from the main branch, loading exactly as you did above, I get an error:
```
BlockSizeError: Got more bytes so far (>5275016) than requested (5251072)
```
My fsspec version is 2021.08.01. I'm going to try with an updated version.

Reply:
@lhoestq after updating `fsspec`, there is a dependency conflict:
- we require `s3fs==2021.08.1`
> s3fs 2021.8.1 requires fsspec==2021.08.1, but you have fsspec 2022.5.0 which is incompatible

Reply:
Checked: with `fsspec==2022.5.0` there is no error and "mlsum" loads:
```
{'text': 'Jean-Jacques Schuhl, Gilles Leroy, Christian Gailly, Yasmina Khadra, James Ellroy, Amos Oz, V. S. Naipaul… Si la rentrée de janvier s\'annonce sous les meilleurs auspices, force est de constater que, avec ...
```

Reply:
I'm investigating the requirement: `s3fs==2021.08.1`

Reply:
@lhoestq I have found the discussion about pinning `s3fs`: https://github.com/huggingface/datasets/pull/2858/files#r700027326

Reply:
I'm going to try to unpin it and see which version pip installs.

Reply:
Also, I have checked which is the minimum version of `fsspec` that does not raise the error: `fsspec>=2021.11.1`. I'm pinning it.

Reply:
When `s3fs` is unpinned, CI installs `s3fs-0.5.1`.